File: baekjoon/Level/Silver/24444. 알고리즘 수업 - 너비 우선 탐색 1/알고리즘 수업 - 너비 우선 탐색 1.swift
func bfs(_ v: Int) {
    var queue: [Int] = [v]
    var idx = 0
    visited[v] = count
    
    while idx < queue.count {
        let n = queue[idx]
        
        for i in graph[n] {
            if visited[i] == 0 {
                queue.append(i)
                count += 1
                visited[i] = count
            }
        }
        idx += 1
    }
}

let input = readLine()!.split(separator: " ").map{ Int($0)! }
let (N, M, R) = (input[0], input[1], input[2])

var graph: [[Int]] = Array(repeating: [], count: N + 1)

for _ in 0..<M {
    let input = readLine()!.split(separator: " ").map{ Int($0)! }
    graph[input[0]].append(input[1])
    graph[input[1]].append(input[0])
}

graph = graph.map{ $0.sorted() }

var visited: [Int] = Array(repeating: 0, count: N + 1)
var count = 1

bfs(R)

_ = visited[1...].map{ print($0) }

File: baekjoon/Level/Silver/24444. 알고리즘 수업 - 너비 우선 탐색 1/알고리즘 수업 - 너비 우선 탐색 1.py
from collections import deque

def bfs(v):
    global count
    queue = deque([v])
    visited[v] = count
    
    while queue:
        n = queue.popleft()
        for i in sorted(graph[n]):
            if visited[i] == 0:
                queue.append(i)
                count += 1
                visited[i] = count

N, M, R = map(int, input().split())

graph = [[] for _ in range(N + 1)]

for _ in range(M):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

visited = [0] * (N + 1)
count = 1

bfs(R)

print("\n".join(map(str, visited[1:])))